resources/external_control.urscript

```diff
@@ -113,11 +146,21 @@ thread servoThread():
       end
 
       q = extrapolate()
-      servoj(q, t=steptime, {{SERVO_J_REPLACE}})
+      if targetWithinLimits(q, steptime):
+        servoj(q, t=steptime, {{SERVO_J_REPLACE}})
+        q_act = get_actual_joint_positions()
+      else:
+        terminateProgram()
+      end
 
     elif state == SERVO_RUNNING:
       extrapolate_count = 0
-      servoj(q, t=steptime, {{SERVO_J_REPLACE}})
+      if targetWithinLimits(q, steptime):
+        servoj(q, t=steptime, {{SERVO_J_REPLACE}})
+        q_act = get_actual_joint_positions()
+      else:
+        terminateProgram()
```

Comment:
I wonder what would be the benefit of actually quitting here? We could instead just ignore the command.